client/me/purchases/cancel-purchase/button.jsx

```diff
@@ -299,7 +298,7 @@ class CancelPurchaseButton extends Component {
 					defaultContent={ this.renderCancellationEffect() }
 					onInputChange={ this.onSurveyChange }
 					purchase={ purchase }
-					selectedSite={ selectedSite }
+					selectedSite={ site }
```

Comment:
I don't think the `site` property actually exists?

This means we're passing a null value to the cancellation form (for all products including WordPress.com), which in turn means that the `selectedSite && <QuerySitePlans siteId={ selectedSite.ID } />` in that component (discussed previously) never runs.

I wasn't entirely sure about the consequences of that, but I found https://github.com/Automattic/wp-calypso/pull/34994/commits/2971dc84f2c23ca50818e470a85e67f4a0f71612 which indicates the `QuerySitePlans` was added recently, for the new WordPress.com plan downgrade functionality in https://github.com/Automattic/wp-calypso/pull/34994.  I couldn't actually figure out how to reproduce a problem if it's not there, but reading that commit it seems like it's probably needed in general.  However, that also implies it's *not* needed for a disconnected Jetpack site.  So I guess it should be safe to just fix this to pass the selected site in when available, but not worry about the fact that it's unavailable for disconnected Jetpack sites.

Reply:
> I don't think the site property actually exists?

This confused me for a bit, because I don't randomly add in variables just to see what happens. I remember having access to this property, and React Dev Tools is showing it's an available property on that component, but where it comes from isn't obvious. If I'm reading the code correctly this property actually comes in [via the `ManagePurchase` component](https://github.com/Automattic/wp-calypso/blob/master/client/me/purchases/manage-purchase/index.jsx#L506) and is used as `selectedSite` anyway. 

Still, this doesn't need to be changed, so I have reverted it. I think the only reason it was changed in the first place was to allow the `ProductLink` component to display and was not reverted when it was clear the link itself wouldn't work because additional refactoring would need to be done on the components `ProductLink` links to. Given what you've pointed out with `QuerySitePlans`, I agree it's best not to disturb this component.